lib/ibm_cloud_sdk_core/base_service.rb

```diff
@@ -0,0 +1,275 @@
+    args[:params].delete_if { |_k, v| v.nil? } if args[:params].instance_of?(Hash)
+    args[:form].delete_if { |_k, v| v.nil? } if args.key?(:form)
+    args.delete_if { |_, v| v.nil? }
+    args[:headers].delete("Content-Type") if args.key?(:form) || args[:json].nil?
+
+    if @username == "apikey" && !@icp_prefix
+      iam_apikey(iam_apikey: @password)
+      @username = nil
+    end
+
+    conn = @conn
+    if !@token_manager.nil?
+      access_token = @token_manager.token
+      args[:headers]["Authorization"] = "Bearer #{access_token}"
+    elsif !@username.nil? && !@password.nil?
+      conn = @conn.basic_auth(user: @username, pass: @password)
+    end
+
+    args[:headers] = args[:headers].merge(@temp_headers) unless @temp_headers.nil?
+    @temp_headers = nil unless @temp_headers.nil?
+
+    if args.key?(:form)
+      response = conn.follow.request(
+        args[:method],
+        HTTP::URI.parse(@url + args[:url]),
+        headers: conn.default_options.headers.merge(HTTP::Headers.coerce(args[:headers])),
+        params: args[:params],
+        form: args[:form]
+      )
+    else
+      response = conn.follow.request(
+        args[:method],
+        HTTP::URI.parse(@url + args[:url]),
+        headers: conn.default_options.headers.merge(HTTP::Headers.coerce(args[:headers])),
+        body: args[:json],
+        params: args[:params]
+      )
+    end
+    return DetailedResponse.new(response: response) if (200..299).cover?(response.code)
+
+    raise ApiException.new(response: response)
+  end
+
+  # @note Chainable
+  # @param headers [Hash] Custom headers to be sent with the request
+  # @return [self]
+  def headers(headers)
+    raise TypeError("Expected Hash type, received #{headers.class}") unless headers.instance_of?(Hash)
+
+    @temp_headers = headers
+    self
+  end
+
+  # @!method configure_http_client(proxy: {}, timeout: {}, disable_ssl: false)
+  # Sets the http client config, currently works with timeout and proxies
+  # @param proxy [Hash] The hash of proxy configurations
+  # @option proxy address [String] The address of the proxy
+  # @option proxy port [Integer] The port of the proxy
+  # @option proxy username [String] The username of the proxy, if authentication is needed
+  # @option proxy password [String] The password of the proxy, if authentication is needed
+  # @option proxy headers [Hash] The headers to be used with the proxy
+  # @param timeout [Hash] The hash for configuring timeouts. `per_operation` has priority over `global`
+  # @option timeout per_operation [Hash] Timeouts per operation. Requires `read`, `write`, `connect`
+  # @option timeout global [Integer] Upper bound on total request time
+  # @param disable_ssl [Boolean] Disable the SSL verification (Note that this has serious security implications - only do this if you really mean to!)
+  def configure_http_client(proxy: {}, timeout: {}, disable_ssl: false)
```

Comment:
We should use `disable_ssl_verification` not `disable_ssl` I think we talked about this :(

Reply:
updating!